base/libgit2/remote.jl

```diff
@@ -88,3 +88,9 @@ function push{T<:AbstractString}(rmt::GitRemote, refspecs::Vector{T};
         !no_refs && finalize(sa)
     end
 end
+
+function name(rmt::GitRemote)
+    str_ptr = ccall((:git_remote_name, :libgit2), Cstring, (Ptr{Void}, ), rmt.ptr)
+    str_ptr == C_NULL && throw(Error.GitError(Error.ERROR))
```

Comment:
No need for exception, return `nothing`, because  C_NULL is returned for in-memory remotes.